src/test/java/uk/gov/pay/adminusers/persistence/dao/UserDaoIT.java

```diff
@@ -349,4 +351,46 @@ public void shouldNotFindAnyUser() {
 
         assertThat(users.isEmpty(), is(true));
     }
+
+    @Test
+    public void shouldNotCreateAUserWithDifferentCaseEmail() {
+        Role existingRole = roleDbFixture(databaseHelper)
+                .withName("admin")
+                .insertRole();
+
+        int serviceId = serviceDbFixture(databaseHelper).insertService().getId();
+        userDbFixture(databaseHelper)
+                .withUsername("user@example.com")
+                .withEmail("user@example.com")
+                .withServiceRole(serviceId, existingRole.getId()).insertUser();
+
+        Role role = roleDbFixture(databaseHelper).insertRole();
+
+        String username = valueOf(nextInt());
+
+        UserEntity userEntity = new UserEntity();
+        userEntity.setExternalId(randomUuid());
+        userEntity.setUsername("User@example.com");
+        userEntity.setPassword("password-" + username);
+        userEntity.setDisabled(false);
+        userEntity.setEmail("User@example.com");
+        userEntity.setOtpKey(randomInt().toString());
+        userEntity.setTelephoneNumber("+447700900000");
+        userEntity.setSecondFactor(SecondFactorMethod.SMS);
+        userEntity.setSessionVersion(0);
+        ZonedDateTime timeNow = ZonedDateTime.now(ZoneId.of("UTC"));
+        userEntity.setCreatedAt(timeNow);
+        userEntity.setUpdatedAt(timeNow);
+
+        Optional<ServiceEntity> serviceEntity = serviceDao.findById(serviceId);
+        RoleEntity roleEntity = roleDao.findByRoleName(role.getName()).get();
+
+        ServiceRoleEntity serviceRoleEntity = new ServiceRoleEntity(serviceEntity.get(), roleEntity);
+        serviceRoleEntity.setUser(userEntity);
+
+        userEntity.addServiceRole(serviceRoleEntity);
+
+        var thrown = assertThrows(javax.persistence.RollbackException.class, () -> userDao.persist(userEntity));
+        assertThat(thrown.getMessage(), containsString("ERROR: duplicate key value violates unique constraint \"lower_case_email_index\""));
```

Comment:
Is the plan to handle this error later in the service/resource layer and return 409?